File: CelesteMapEditor/src/player_inventory.cpp
#include "player_inventory.hpp"

const celeste::PlayerInventory celeste::PlayerInventory::Prologue(0, false);
const celeste::PlayerInventory celeste::PlayerInventory::Default;
const celeste::PlayerInventory celeste::PlayerInventory::OldSite(1, false);
const celeste::PlayerInventory celeste::PlayerInventory::CH6End(2);
const celeste::PlayerInventory celeste::PlayerInventory::TheSummit(2, true, false);
const celeste::PlayerInventory celeste::PlayerInventory::Core(2, true, true, true);
const celeste::PlayerInventory celeste::PlayerInventory::Farewell(1, true, false);

celeste::PlayerInventory::PlayerInventory(const int dashes, const bool dreamDash, const bool backpack, const bool noRefills)
	: dashes(dashes), dreamDash(dreamDash), backpack(backpack), noRefills(noRefills)
{
}

File: MapEditor/Editor/Celeste/PlayerInventory.cs
namespace Editor.Celeste
{
    public struct PlayerInventory
    {
        public static readonly PlayerInventory Prologue = new() { Dashes = 0, DreamDash = false };
        public static readonly PlayerInventory Default = new();
        public static readonly PlayerInventory OldSite = new() { DreamDash = false };
        public static readonly PlayerInventory CH6End = new() { Dashes = 2 };
        public static readonly PlayerInventory TheSummit = new() { Dashes = 2, DreamDash = true, Backpack = false };
        public static readonly PlayerInventory Core = new() { Dashes = 2, NoRefills = true };
        public static readonly PlayerInventory Farewell = new() { Backpack = false };

        public int Dashes = 1;
        public bool DreamDash = true;
        public bool Backpack = true;
        public bool NoRefills = false;

        public PlayerInventory()
        {
        }

        public static PlayerInventory Parse(string value)
        {
            return value switch
            {
                "Prologue" => Prologue,
                "Default" => Default,
                "OldSite" => OldSite,
                "CH6End" => CH6End,
                "TheSummit" => TheSummit,
                "Core" => Core,
                "Farewell" => Farewell,
                _ => Default,
            };
        }
    }
}
